Created Allergies Tab

itsisak opened the pull request:
As discussed previously, I have now created a seperate tab for allergies in EventAdministrate. Currently anyone can still see all the allergies, but we can add an anonymized list later on.  

Before:

![Skjermbilde 2023-02-12 kl  00 56 11](https://user-images.githubusercontent.com/45620425/218286435-f834412f-c230-4dce-93d9-1d417358738d.jpeg)

After:
![Skjermbilde 2023-02-12 kl  00 56 35](https://user-images.githubusercontent.com/45620425/218286468-909a6e06-008c-4f39-9385-013503ddf88a.jpeg)
![Skjermbilde 2023-02-12 kl  00 57 49](https://user-images.githubusercontent.com/45620425/218286473-55fa9131-7d2b-4116-8bf3-e13081ac9115.jpeg)


Newst version:
<img width="1102" alt="Skjermbilde 2023-02-21 kl  22 04 20" src="https://user-images.githubusercontent.com/45620425/220458723-abac8d50-4814-4408-8cb2-eacdf3b4db3a.png">



Most of the code in Allergies.tsx and EventAllergiesRoute.ts was copied and slightly altered from Attendees.tsx and EventAttendees.ts. Unsure if everything is necessary or if there is anything I should have done differently, please let me know if so. 


Resolves ABA-175

haukkagu reviewed (approved):
LGTM

itsisak commented:
Also, I added "Status" and "Til stede" to the Allergies tab, however Im unsure if this is usefull. I figured it might come in handy, put let me know and I can remove it as it is already in the "påmeldinger" tab.

PeterJFB reviewed (approved):
LGTM :100:

ivarnakken commented:
> Also, I added "Status" and "Til stede" to the Allergies tab, however Im unsure if this is usefull. I figured it might come in handy, put let me know and I can remove it as it is already in the "påmeldinger" tab.

@itsisak 
I don't really think we need status.

erlingfn commented on app/routes/events/components/EventAdministrate/Allergies.tsx:
```diff
@@ -0,0 +1,155 @@
+  loading: boolean;
+  registered: Array<EventRegistration>;
+  unregistered: Array<EventRegistration>;
+  unregister: (arg0: {
```
Is it possible to get a better name than arg0?

itsisak replied:
I just now pushed a new change where I removed both the unregister and updatePresence (comment below) all together as they are not used in the allergies tab.

itsisak replied:
I have now fixed these issues in the original Attendes file as well as some deprecated syntax

erlingfn commented on app/routes/events/components/EventAdministrate/Allergies.tsx:
```diff
@@ -0,0 +1,155 @@
+    registrationId: ID;
+    admin: boolean;
+  }) => Promise<void>;
+  updatePresence: (arg0: number, arg1: number, arg2: string) => Promise<any>;
```
Same here. I know you just copied this from another place, but I think this is an excellent chance to improve it <3

erlingfn commented:
>  I don't really think we need status.

It agree. Does not hurt, but cannot really see how it is useful

erlingfn reviewed (changes requested):
Looks good! Would be great if you improve the types when you still remember how this works, and I think you can remove the presence, but otherwise this is great.

itsisak commented:
I removed the "Til stede" column, but I kept the "Status" and added sorting functionality to it, it seems nice to know if the allergies listed are for people that are actually registered or just on the waiting list.

haukkagu reviewed (approved):
LGTM💯

itsisak commented:
Fixed yarn lint
